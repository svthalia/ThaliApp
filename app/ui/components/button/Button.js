import React from 'react';
import { Platform, Text, TouchableHighlight, View, ViewPropTypes } from 'react-native';
import PropTypes from 'prop-types';
import styles from './style/Button';
import Colors from '../../style/Colors';

const Button = (props) => {
  const touchableStyles = [props.style, styles.touchable];
  if (props.disabled) {
    touchableStyles.push(styles.disabled);
  }
  const title = Platform.OS === 'ios' ? props.title : props.title.toUpperCase();
  return (
    <TouchableHighlight
      disabled={props.disabled}
      onPress={props.disabled ? null : props.onPress}
      style={touchableStyles}
      underlayColor={props.underlayColor}
    >
      <View
        style={[
          styles.container,
          props.containerStyle,
          { backgroundColor: props.color },
        ]}
      >
        <Text style={[styles.text, props.textStyle]}>{title}</Text>
      </View>
    </TouchableHighlight>
  );
};

Button.propTypes = {
  ...ViewPropTypes,
  color: PropTypes.string,
  disabled: PropTypes.bool,
  onPress: PropTypes.func.isRequired,
  containerStyle: Text.propTypes.style,
  // eslint-disable-next-line react/no-typos
  textStyle: Text.propTypes.style,
  title: PropTypes.string.isRequired,
  underlayColor: PropTypes.string,
};

Button.defaultProps = {
  color: Colors.magenta,
  disabled: false,
  containerStyle: {},
  textStyle: {},
  underlayColor: Colors.white,
};

export default Button;
