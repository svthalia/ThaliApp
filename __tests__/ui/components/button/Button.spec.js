import React from 'react';
import { Platform } from 'react-native';
import renderer from 'react-test-renderer';
import Button from '../../../../app/ui/components/button/Button';

describe('Button component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button title='title' onPress={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders disabled correctly', () => {
    const tree = renderer
      .create(<Button title='title' disabled onPress={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders title correctly on Android', () => {
    Platform.OS = 'android';
    const tree = renderer.create(<Button title='title' onPress={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
