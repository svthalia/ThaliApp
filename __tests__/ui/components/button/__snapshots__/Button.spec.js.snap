// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      undefined,
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "padding": 10,
        },
        Object {},
        Object {
          "backgroundColor": "#E62272",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 14,
            "justifyContent": "center",
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      TITLE
    </Text>
  </View>
</View>
`;

exports[`Button component renders disabled correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      undefined,
      Object {},
      Object {
        "opacity": 0.8,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "padding": 10,
        },
        Object {},
        Object {
          "backgroundColor": "#E62272",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 14,
            "justifyContent": "center",
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      TITLE
    </Text>
  </View>
</View>
`;

exports[`Button component renders title correctly on Android 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      undefined,
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "padding": 10,
        },
        Object {},
        Object {
          "backgroundColor": "#E62272",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 14,
            "justifyContent": "center",
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      TITLE
    </Text>
  </View>
</View>
`;
